lib/imports/validator.js

```diff
@@ -0,0 +1,40 @@
+import validator from 'jsen'
+import fieldValidation from './field-validation'
+
+export function validateDocField({ doc, field, newValue }) {
+  var data = doc.content || doc.facts
+  if (!data) throw new Meteor.Error('No valid data in document')
+  var docSchemaKey = doc.schema[field]
+  var schemaVal = {
+    type: 'string',
+    required: true
+  }
+  if (docSchemaKey) {
+    Object.assign(schemaVal, fieldValidation[docSchemaKey.type] || {})
+    Object.assign(schemaVal, docSchemaKey.validation || {})
+  }
+  var miniSchema = {
+    required: true,
+    type: 'object',
+    properties: { [field]: schemaVal },
+    additionalProperties: false
+  }
+  var validate = validator(miniSchema)
+  var valid = validate({ [field]: newValue })
+  if (!valid) return {
+    message: validationError(validate.errors[0].keyword)
+  }
+}
```

Comment:
I'm finding it difficult to get my head around `validateDocField` that returns a truthy value when invalid.

Joi takes the approach of returning a result object that has either an `error` property (in case of a validation error) or a `value` property which contains the validated (and sometimes normalised) value.

Also the `isInvalid` function I've seen in the react classes seems that it could be more understandable if it was more simply `isValid`.

What do you think?